File: design-tokens.kt
// Design Tokens - Kotlin Constants
object DesignTokens {

    // MARK: - Colors
    @JvmStatic
    val PrimaryLight = Color(0xFF2563EB)

    @JvmStatic
    val PrimaryDark = Color(0xFF3B82F6)

    @JvmStatic
    fun Primary(context: Context): Color {
        return if (isDarkMode(context)) PrimaryDark else PrimaryLight
    }

    @JvmStatic
    val PrimaryPressedLight = Color(0xFF1D4ED8)

    @JvmStatic
    val PrimaryPressedDark = Color(0xFF2563EB)

    @JvmStatic
    fun PrimaryPressed(context: Context): Color {
        return if (isDarkMode(context)) PrimaryPressedDark else PrimaryPressedLight
    }

    @JvmStatic
    val PrimaryActiveLight = Color(0xFF1E40AF)

    @JvmStatic
    val PrimaryActiveDark = Color(0xFF1D4ED8)

    @JvmStatic
    fun PrimaryActive(context: Context): Color {
        return if (isDarkMode(context)) PrimaryActiveDark else PrimaryActiveLight
    }

    @JvmStatic
    val PrimaryDisabledLight = Color(0xFF93C5FD)

    @JvmStatic
    val PrimaryDisabledDark = Color(0xFF1E3A8A)

    @JvmStatic
    fun PrimaryDisabled(context: Context): Color {
        return if (isDarkMode(context)) PrimaryDisabledDark else PrimaryDisabledLight
    }

    @JvmStatic
    val SecondaryLight = Color(0xFFF3F4F6)

    @JvmStatic
    val SecondaryDark = Color(0xFF374151)

    @JvmStatic
    fun Secondary(context: Context): Color {
        return if (isDarkMode(context)) SecondaryDark else SecondaryLight
    }

    @JvmStatic
    val SecondaryPressedLight = Color(0xFFE5E7EB)

    @JvmStatic
    val SecondaryPressedDark = Color(0xFF4B5563)

    @JvmStatic
    fun SecondaryPressed(context: Context): Color {
        return if (isDarkMode(context)) SecondaryPressedDark else SecondaryPressedLight
    }

    @JvmStatic
    val BgColorLight = Color(0xFFFFFFFF)

    @JvmStatic
    val BgColorDark = Color(0xFF0F172A)

    @JvmStatic
    fun BgColor(context: Context): Color {
        return if (isDarkMode(context)) BgColorDark else BgColorLight
    }

    @JvmStatic
    val SurfaceLight = Color(0xFFF9FAFB)

    @JvmStatic
    val SurfaceDark = Color(0xFF1E293B)

    @JvmStatic
    fun Surface(context: Context): Color {
        return if (isDarkMode(context)) SurfaceDark else SurfaceLight
    }

    @JvmStatic
    val PrimaryTextLight = Color(0xFF111827)

    @JvmStatic
    val PrimaryTextDark = Color(0xFFF8FAFC)

    @JvmStatic
    fun PrimaryText(context: Context): Color {
        return if (isDarkMode(context)) PrimaryTextDark else PrimaryTextLight
    }

    @JvmStatic
    val SecondaryTextLight = Color(0xFF6B7280)

    @JvmStatic
    val SecondaryTextDark = Color(0xFFCBD5E1)

    @JvmStatic
    fun SecondaryText(context: Context): Color {
        return if (isDarkMode(context)) SecondaryTextDark else SecondaryTextLight
    }

    @JvmStatic
    val TextDisabledLight = Color(0xFF9CA3AF)

    @JvmStatic
    val TextDisabledDark = Color(0xFF64748B)

    @JvmStatic
    fun TextDisabled(context: Context): Color {
        return if (isDarkMode(context)) TextDisabledDark else TextDisabledLight
    }

    @JvmStatic
    val OnPrimaryTextLight = Color(0xFFFFFFFF)

    @JvmStatic
    val OnPrimaryTextDark = Color(0xFFFFFFFF)

    @JvmStatic
    fun OnPrimaryText(context: Context): Color {
        return if (isDarkMode(context)) OnPrimaryTextDark else OnPrimaryTextLight
    }

    @JvmStatic
    val CardLight = Color(0xFFFFFFFF)

    @JvmStatic
    val CardDark = Color(0xFF1E293B)

    @JvmStatic
    fun Card(context: Context): Color {
        return if (isDarkMode(context)) CardDark else CardLight
    }

    @JvmStatic
    val DividerLight = Color(0xFFE5E7EB)

    @JvmStatic
    val DividerDark = Color(0xFF334155)

    @JvmStatic
    fun Divider(context: Context): Color {
        return if (isDarkMode(context)) DividerDark else DividerLight
    }

    @JvmStatic
    val ErrorLight = Color(0xFFDC2626)

    @JvmStatic
    val ErrorDark = Color(0xFFEF4444)

    @JvmStatic
    fun Error(context: Context): Color {
        return if (isDarkMode(context)) ErrorDark else ErrorLight
    }

    @JvmStatic
    val SuccessLight = Color(0xFF16A34A)

    @JvmStatic
    val SuccessDark = Color(0xFF22C55E)

    @JvmStatic
    fun Success(context: Context): Color {
        return if (isDarkMode(context)) SuccessDark else SuccessLight
    }

    @JvmStatic
    val WarningLight = Color(0xFFD97706)

    @JvmStatic
    val WarningDark = Color(0xFFF59E0B)

    @JvmStatic
    fun Warning(context: Context): Color {
        return if (isDarkMode(context)) WarningDark else WarningLight
    }

    @JvmStatic
    val InfoLight = Color(0xFF0EA5E9)

    @JvmStatic
    val InfoDark = Color(0xFF06B6D4)

    @JvmStatic
    fun Info(context: Context): Color {
        return if (isDarkMode(context)) InfoDark else InfoLight
    }

    // MARK: - Typography
    // Font Families
    @JvmStatic
    val MainFont = "Inter"

    @JvmStatic
    val DisplayFont = "Poppins"

    // Font Weights
    @JvmStatic
    val FontWeightLight = FontWeight.w300

    @JvmStatic
    val FontWeightRegular = FontWeight.w400

    @JvmStatic
    val FontWeightMedium = FontWeight.w500

    @JvmStatic
    val FontWeightSemiBold = FontWeight.w600

    @JvmStatic
    val FontWeightBold = FontWeight.w700

    // Font Sizes
    @JvmStatic
    val FontSizeXS = 12.sp

    @JvmStatic
    val FontSizeS = 14.sp

    @JvmStatic
    val FontSizeMD = 16.sp

    @JvmStatic
    val FontSizeLG = 18.sp

    @JvmStatic
    val FontSizeXL = 20.sp

    @JvmStatic
    val FontSizeXXL = 24.sp

    @JvmStatic
    val FontSizeXXXL = 32.sp

    @JvmStatic
    val FontSizeDisplay = 48.sp

    // Line Heights
    @JvmStatic
    val LineHeightTight = 1.25

    @JvmStatic
    val LineHeightNormal = 1.5

    @JvmStatic
    val LineHeightRelaxed = 1.625

    @JvmStatic
    val LineHeightLoose = 2

    // Letter Spacings
    @JvmStatic
    val LetterSpacingTight = -0.002.em

    @JvmStatic
    val LetterSpacingNormal = 0.000.em

    @JvmStatic
    val LetterSpacingWide = 0.002.em

    @JvmStatic
    val LetterSpacingWider = 0.003.em

    // MARK: - Spacing
    @JvmStatic
    val SpacingXS = 4.dp

    @JvmStatic
    val SpacingS = 8.dp

    @JvmStatic
    val SpacingMD = 16.dp

    @JvmStatic
    val SpacingLG = 24.dp

    @JvmStatic
    val SpacingXL = 32.dp

    @JvmStatic
    val SpacingXXL = 40.dp

    @JvmStatic
    val SpacingXXXL = 48.dp

    @JvmStatic
    val SpacingXXXXL = 64.dp

    // MARK: - Sizing
    @JvmStatic
    val SizingXS = 24.dp

    @JvmStatic
    val SizingS = 32.dp

    @JvmStatic
    val SizingMD = 40.dp

    @JvmStatic
    val SizingLG = 48.dp

    @JvmStatic
    val SizingXL = 56.dp

    @JvmStatic
    val SizingXXL = 64.dp

    @JvmStatic
    val SizingXXXL = 80.dp

    @JvmStatic
    val HeightButton = 44.dp

    @JvmStatic
    val HeightInput = 48.dp

    @JvmStatic
    val HeightCard = 120.dp

    @JvmStatic
    val WidthSidebar = 280.dp

    @JvmStatic
    val WidthModal = 400.dp

    @JvmStatic
    val WidthContainer = 1200.dp

    @JvmStatic
    val IconSizeXS = 12.dp

    @JvmStatic
    val IconSizeS = 16.dp

    @JvmStatic
    val IconSizeMD = 20.dp

    @JvmStatic
    val IconSizeLG = 24.dp

    @JvmStatic
    val IconSizeXL = 32.dp

    // MARK: - Border Radius
    @JvmStatic
    val BorderRadiusXS = 4.dp

    @JvmStatic
    val BorderRadiusS = 8.dp

    @JvmStatic
    val BorderRadiusMD = 12.dp

    @JvmStatic
    val BorderRadiusLG = 16.dp

    @JvmStatic
    val BorderRadiusXL = 20.dp

    @JvmStatic
    val BorderRadiusXXL = 24.dp

    @JvmStatic
    val BorderRadiusFull = 9999.dp

    // MARK: - Borders
    @JvmStatic
    val BorderThin = 1.dp

    @JvmStatic
    val BorderMedium = 1.5.dp

    @JvmStatic
    val BorderThick = 2.dp

    @JvmStatic
    val BorderFocus = 3.dp

    // MARK: - Shadows
    // Small Shadows
    @JvmStatic
    val BoxShadowS = Shadow(
        color = Color(0xFF000000).copy(alpha = 0.25f),
        offset = Offset(0.dp, 1.dp),
        blurRadius = 3.dp
    )

    // Medium Shadows
    @JvmStatic
    val BoxShadowMD = Shadow(
        color = Color(0xFF000000).copy(alpha = 0.25f),
        offset = Offset(0.dp, 4.dp),
        blurRadius = 6.dp
    )

    // Large Shadows
    @JvmStatic
    val BoxShadowLG = Shadow(
        color = Color(0xFF000000).copy(alpha = 0.25f),
        offset = Offset(0.dp, 10.dp),
        blurRadius = 15.dp
    )

    // MARK: - Opacities
    @JvmStatic
    val ShadowOpacity = 0.1.f

    @JvmStatic
    val OverlayOpacity = 0.6.f

    @JvmStatic
    val DisabledOpacity = 0.4.f

    // MARK: - Z-Indices
    @JvmStatic
    val ZIndexDropdown = 1000

    @JvmStatic
    val ZIndexModal = 1050

    @JvmStatic
    val ZIndexTooltip = 1100

    // MARK: - Transitions
    @JvmStatic
    val TransitionFast = 150

    @JvmStatic
    val TransitionNormal = 200

    @JvmStatic
    val TransitionSlow = 300

    // MARK: - Utility Functions
    @JvmStatic
    fun isDarkMode(context: Context): Boolean {
        val configuration = context.resources.configuration
        val currentNightMode = configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
        return currentNightMode == Configuration.UI_MODE_NIGHT_YES
    }

    @JvmStatic
    fun getThemeColors(context: Context): Colors {
        return if (isDarkMode(context)) darkColors else lightColors
    }

    @JvmStatic
    fun getTextColor(context: Context): Color {
        return if (isDarkMode(context)) Color.White else Color.Black
    }
}

File: design-tokens.swift
// Design Tokens - SwiftUI Constants
import SwiftUI

public enum DesignTokens {

    // MARK: - Colors
    static let PrimaryLight = Color(red: 1, green: 0.1450980392156863, blue: 0.38823529411764707)

    static let PrimaryDark = Color(red: 1, green: 0.23137254901960785, blue: 0.5098039215686274)

    static func Primary(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? PrimaryDark : PrimaryLight
    }

    static let PrimaryPressedLight = Color(red: 1, green: 0.11372549019607843, blue: 0.3058823529411765)

    static let PrimaryPressedDark = Color(red: 1, green: 0.1450980392156863, blue: 0.38823529411764707)

    static func PrimaryPressed(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? PrimaryPressedDark : PrimaryPressedLight
    }

    static let PrimaryActiveLight = Color(red: 1, green: 0.11764705882352941, blue: 0.25098039215686274)

    static let PrimaryActiveDark = Color(red: 1, green: 0.11372549019607843, blue: 0.3058823529411765)

    static func PrimaryActive(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? PrimaryActiveDark : PrimaryActiveLight
    }

    static let PrimaryDisabledLight = Color(red: 1, green: 0.5764705882352941, blue: 0.7725490196078432)

    static let PrimaryDisabledDark = Color(red: 1, green: 0.11764705882352941, blue: 0.22745098039215686)

    static func PrimaryDisabled(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? PrimaryDisabledDark : PrimaryDisabledLight
    }

    static let SecondaryLight = Color(red: 1, green: 0.9529411764705882, blue: 0.9568627450980393)

    static let SecondaryDark = Color(red: 1, green: 0.21568627450980393, blue: 0.2549019607843137)

    static func Secondary(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? SecondaryDark : SecondaryLight
    }

    static let SecondaryPressedLight = Color(red: 1, green: 0.8980392156862745, blue: 0.9058823529411765)

    static let SecondaryPressedDark = Color(red: 1, green: 0.29411764705882354, blue: 0.3333333333333333)

    static func SecondaryPressed(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? SecondaryPressedDark : SecondaryPressedLight
    }

    static let BgColorLight = Color(red: 1, green: 1, blue: 1)

    static let BgColorDark = Color(red: 1, green: 0.058823529411764705, blue: 0.09019607843137255)

    static func BgColor(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? BgColorDark : BgColorLight
    }

    static let SurfaceLight = Color(red: 1, green: 0.9764705882352941, blue: 0.9803921568627451)

    static let SurfaceDark = Color(red: 1, green: 0.11764705882352941, blue: 0.1607843137254902)

    static func Surface(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? SurfaceDark : SurfaceLight
    }

    static let PrimaryTextLight = Color(red: 1, green: 0.06666666666666667, blue: 0.09411764705882353)

    static let PrimaryTextDark = Color(red: 1, green: 0.9725490196078431, blue: 0.9803921568627451)

    static func PrimaryText(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? PrimaryTextDark : PrimaryTextLight
    }

    static let SecondaryTextLight = Color(red: 1, green: 0.4196078431372549, blue: 0.4470588235294118)

    static let SecondaryTextDark = Color(red: 1, green: 0.796078431372549, blue: 0.8352941176470589)

    static func SecondaryText(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? SecondaryTextDark : SecondaryTextLight
    }

    static let TextDisabledLight = Color(red: 1, green: 0.611764705882353, blue: 0.6392156862745098)

    static let TextDisabledDark = Color(red: 1, green: 0.39215686274509803, blue: 0.4549019607843137)

    static func TextDisabled(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? TextDisabledDark : TextDisabledLight
    }

    static let OnPrimaryTextLight = Color(red: 1, green: 1, blue: 1)

    static let OnPrimaryTextDark = Color(red: 1, green: 1, blue: 1)

    static func OnPrimaryText(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? OnPrimaryTextDark : OnPrimaryTextLight
    }

    static let CardLight = Color(red: 1, green: 1, blue: 1)

    static let CardDark = Color(red: 1, green: 0.11764705882352941, blue: 0.1607843137254902)

    static func Card(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? CardDark : CardLight
    }

    static let DividerLight = Color(red: 1, green: 0.8980392156862745, blue: 0.9058823529411765)

    static let DividerDark = Color(red: 1, green: 0.2, blue: 0.2549019607843137)

    static func Divider(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? DividerDark : DividerLight
    }

    static let ErrorLight = Color(red: 1, green: 0.8627450980392157, blue: 0.14901960784313725)

    static let ErrorDark = Color(red: 1, green: 0.9372549019607843, blue: 0.26666666666666666)

    static func Error(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? ErrorDark : ErrorLight
    }

    static let SuccessLight = Color(red: 1, green: 0.08627450980392157, blue: 0.6392156862745098)

    static let SuccessDark = Color(red: 1, green: 0.13333333333333333, blue: 0.7725490196078432)

    static func Success(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? SuccessDark : SuccessLight
    }

    static let WarningLight = Color(red: 1, green: 0.8509803921568627, blue: 0.4666666666666667)

    static let WarningDark = Color(red: 1, green: 0.9607843137254902, blue: 0.6196078431372549)

    static func Warning(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? WarningDark : WarningLight
    }

    static let InfoLight = Color(red: 1, green: 0.054901960784313725, blue: 0.6470588235294118)

    static let InfoDark = Color(red: 1, green: 0.023529411764705882, blue: 0.7137254901960784)

    static func Info(colorScheme: ColorScheme) -> Color {
        return colorScheme == .dark ? InfoDark : InfoLight
    }

    // MARK: - Typography
    // Font Families
    static let MainFont = "Inter"

    static let DisplayFont = "Poppins"

    // Font Weights
    static let FontWeightLight = Font.Weight.light

    static let FontWeightRegular = Font.Weight.regular

    static let FontWeightMedium = Font.Weight.medium

    static let FontWeightSemiBold = Font.Weight.semibold

    static let FontWeightBold = Font.Weight.bold

    // Font Sizes
    static let FontSizeXS = CGFloat(12)

    static let FontSizeS = CGFloat(14)

    static let FontSizeMD = CGFloat(16)

    static let FontSizeLG = CGFloat(18)

    static let FontSizeXL = CGFloat(20)

    static let FontSizeXXL = CGFloat(24)

    static let FontSizeXXXL = CGFloat(32)

    static let FontSizeDisplay = CGFloat(48)

    // Line Heights
    static let LineHeightTight = CGFloat(1.25)

    static let LineHeightNormal = CGFloat(1.5)

    static let LineHeightRelaxed = CGFloat(1.625)

    static let LineHeightLoose = CGFloat(2)

    // Letter Spacings
    static let LetterSpacingTight = CGFloat(-0.002)

    static let LetterSpacingNormal = CGFloat(0.000)

    static let LetterSpacingWide = CGFloat(0.002)

    static let LetterSpacingWider = CGFloat(0.003)

    // MARK: - Spacing
    static let SpacingXS = CGFloat(4)

    static let SpacingS = CGFloat(8)

    static let SpacingMD = CGFloat(16)

    static let SpacingLG = CGFloat(24)

    static let SpacingXL = CGFloat(32)

    static let SpacingXXL = CGFloat(40)

    static let SpacingXXXL = CGFloat(48)

    static let SpacingXXXXL = CGFloat(64)

    // MARK: - Sizing
    static let SizingXS = CGFloat(24)

    static let SizingS = CGFloat(32)

    static let SizingMD = CGFloat(40)

    static let SizingLG = CGFloat(48)

    static let SizingXL = CGFloat(56)

    static let SizingXXL = CGFloat(64)

    static let SizingXXXL = CGFloat(80)

    static let HeightButton = CGFloat(44)

    static let HeightInput = CGFloat(48)

    static let HeightCard = CGFloat(120)

    static let WidthSidebar = CGFloat(280)

    static let WidthModal = CGFloat(400)

    static let WidthContainer = CGFloat(1200)

    static let IconSizeXS = CGFloat(12)

    static let IconSizeS = CGFloat(16)

    static let IconSizeMD = CGFloat(20)

    static let IconSizeLG = CGFloat(24)

    static let IconSizeXL = CGFloat(32)

    // MARK: - Border Radius
    static let BorderRadiusXS = CGFloat(4)

    static let BorderRadiusS = CGFloat(8)

    static let BorderRadiusMD = CGFloat(12)

    static let BorderRadiusLG = CGFloat(16)

    static let BorderRadiusXL = CGFloat(20)

    static let BorderRadiusXXL = CGFloat(24)

    static let BorderRadiusFull = CGFloat(9999)

    // MARK: - Borders
    static let BorderThin = CGFloat(1)

    static let BorderMedium = CGFloat(1.5)

    static let BorderThick = CGFloat(2)

    static let BorderFocus = CGFloat(3)

    // MARK: - Shadows
    // Small Shadows
    static let BoxShadowS = Shadow(
        color: Color(red: 1, green: 0, blue: 0, opacity: 0.25),
        radius: 3,
        x: 0,
        y: 1
    )

    // Medium Shadows
    static let BoxShadowMD = Shadow(
        color: Color(red: 1, green: 0, blue: 0, opacity: 0.25),
        radius: 6,
        x: 0,
        y: 4
    )

    // Large Shadows
    static let BoxShadowLG = Shadow(
        color: Color(red: 1, green: 0, blue: 0, opacity: 0.25),
        radius: 15,
        x: 0,
        y: 10
    )

    // MARK: - Opacities
    static let ShadowOpacity = CGFloat(0.1)

    static let OverlayOpacity = CGFloat(0.6)

    static let DisabledOpacity = CGFloat(0.4)

    // MARK: - Z-Indices
    static let ZIndexDropdown = CGFloat(1000)

    static let ZIndexModal = CGFloat(1050)

    static let ZIndexTooltip = CGFloat(1100)

    // MARK: - Transitions
    static let TransitionFast = CGFloat(150)

    static let TransitionNormal = CGFloat(200)

    static let TransitionSlow = CGFloat(300)

    // MARK: - Utility Functions
    static func isDarkMode(colorScheme: ColorScheme) -> Bool {
        return colorScheme == .dark
    }

    static func getTextColor(colorScheme: ColorScheme) -> Color {
        return isDarkMode(colorScheme: colorScheme) ? .white : .black
    }
}

